Extensions/timestamps.js

```diff
@@ -224,12 +253,88 @@ XKit.extensions.timestamps = new Object({
 
 			var timestamp = responseData.posts[0].timestamp;
 			date_element.html(this.format_date(moment(new Date(timestamp * 1000))));
-			date_element.removeClass("xtimestamp_loading");
+			date_element.removeClass("xtimestamp-loading");
 			XKit.storage.set("timestamps", "xkit_timestamp_cache_" + post_id, timestamp);
 		})
 		.catch(() => this.show_failed(date_element));
 	},
 
+	react_add_timestamps: function() {
+		var posts = $("[data-id]:not(.xkit_timestamps)");
```

Comment:
probably best to immediately add the `xkit_timestamps` class once we've collected them for processing
```suggestion
		var posts = $("[data-id]:not(.xkit_timestamps)");
		posts.addClass("xkit_timestamps");
```